source/CINetworkMessageManager.cpp

```diff
@@ -129,14 +130,19 @@ void NetworkMessageManager::receiveMessages() {
                 _gameUpdateManager->addGameUpdate(gameUpdate);
             }
             else if (message_type == NetworkUtils::MessageType::PlanetUpdate) {
-                int srcPlayer = NetworkUtils::decodeInt(recv[1], recv[2], recv[3], recv[4]);
-                int planetColor = NetworkUtils::decodeInt(recv[5], recv[6], recv[7], recv[8]);
-                float planetSize = NetworkUtils::decodeFloat(recv[9], recv[10], recv[11], recv[12]);
-                int timeStamp = NetworkUtils::decodeInt(recv[13], recv[14], recv[15], recv[16]);
+                int srcPlayer = NetworkUtils::decodeInt(recv[4], recv[5], recv[6], recv[7]);
+                int planetColor = NetworkUtils::decodeInt(recv[8], recv[9], recv[10], recv[11]);
+                float planetSize = NetworkUtils::decodeInt(recv[12], recv[13], recv[14], recv[15]);
+                int timestamp = NetworkUtils::decodeFloat(recv[16], recv[17], recv[18], recv[19]);
                 
                 CULog("RCVD PU> SRC[%i], CLR[%i], SIZE[%f]", srcPlayer, planetColor, planetSize);
 
-                // TODO: create opponent planet model when that is complete
+                std::shared_ptr<OpponentPlanet> planet = OpponentPlanet::alloc(0, 0, static_cast<CIColor::Value>(planetColor));
+                planet->setLayerSize(planetSize);
+                std::map<int, std::vector<std::shared_ptr<StardustModel>>> map = {};
+                std::shared_ptr<GameUpdate> gameUpdate = GameUpdate::alloc(_conn->getRoomID(), srcPlayer, map, planet, timestamp);
```

Comment:
should we declare `planet` and `gameUpdate` outside so as to not use up as much memory that won't ever be used later?

Reply:
Yeah I looked into this and its kind of weird to do because of how c++ callbacks work. We'd need to make sure these variables get captured in the callback. For now I think this is fine though